samples/csharp_dotnetcore/07.using-adaptive-cards/Bots/AdaptiveCardsBot.cs

```diff
@@ -48,7 +45,7 @@ protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersA
         protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
         {
             Random r = new Random();
-            var cardAttachment = CreateAdaptiveCardAttachment(this._cards[r.Next(this._cards.Length)]);
+            var cardAttachment = CreateAdaptiveCardAttachment(_cards[r.Next(_cards.Length)]);
```

Comment:
ASP.net idiomatic question.  Why are the property changes made earlier removing the _variableNaming with VariableNaming, but _cards is not Cards?

Reply:
Hi, protected fields, properties and methods are visible to subclases and should be PascalCase based on .Net naming conventions.
In this case, chart is private and only visible to the class, there are no official guidance on capitalization for private fields, some people use camelCase and then you have to use this.myVariable in code to avoid confusing them with local vars, some others (like me and it seems the rest of the bot framework) use underscores to differentiate local variables from privates and to having to avoid to type this.myVariable.